sympy/physics/units/quantities.py

```diff
@@ -89,7 +89,7 @@ def _eval_is_constant(self):
         return self.scale_factor.is_constant()
 
     def _eval_Abs(self):
-        return self.__class__(self.name, self.dimension,
+        return Quantity(self.name, self.dimension,
```

Comment:
@Upabjojr @asmeurer please note that I have disabled edits from maintainers as I would like to understand the reason behind using `self.func`. I don't appreciate changes to the code while we are discussing a problem. Thank you for your understanding.